src/main.cpp

```diff
@@ -1,9 +1,133 @@
-#include <cstdlib>
+#include <algorithm>
+#include <iostream>
+#include <iterator>
 #include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <iomanip>
+#include <string>
+#include <vector>
+using namespace std;
+
+// Global Variables & Functions
+vector<string> all_types = {"PROGRAM", "VAR", "END_VAR", "END_PROGRAM", "BOOL", "TRUE", "FALSE"}; //keywords
+vector<string> operators = {"<", ">", "+", "-", "%", ":="}; // operators
+vector<string> all_sc = {"{", "}", "[", "]", ";", ":"}; // special characters
+void tokenizerFunc (string sample_input); // main lexer function
+typedef struct Token{ // where/how the keywords are going to be saved
+    string key;
+    string value;
+}TOTO;
+vector<TOTO> fin_tokens;
+void display (Token token); //display function 
 
 int main(int argc, char *argv[])
 {
-    std::cout << "argc == " << argc << '\n';
+    // Passing in the program file & modifiying it 
+    ifstream inFile;
+    inFile.open("test2.txt"); // Keep in mind program file directory 
+    stringstream strFile;
+    strFile << inFile.rdbuf();
+
+    string fileString = strFile.str(); // file is now a long string stored as fileString
+    replace( fileString.begin(), fileString.end(), '\n', ' ' ); // find newlines & replace with space
+
+    //call main lexer function & display contents
+    tokenizerFunc(fileString); 
+    cout << "Key \t\t    value" << endl;
+    cout << "----------------------------------------" << endl;
+    for(int i = 0; i < fin_tokens.size(); i++){
+        display(fin_tokens.at(i));
+    }
 
     return EXIT_SUCCESS;
+}
+void display (Token token){
```

Comment:
It looks like this function pretty-prints a Token instance so that it can be displayed in the debug logs. If so, please make this function a public method of the Token class. Rather than writing `display(myToken);`, you should be able to write `std::cout << myToken.to_string() << std::endl`.

Reply:
I've created a method & displayed contents differently.